src/h3lib/lib/localij.c

```diff
@@ -0,0 +1,544 @@
+ *
+ * Failure may occur if the index is too far away from the origin
+ * or if the index is on the other side of a pentagon.
+ *
+ * This function is experimental, and its output is not guaranteed
+ * to be compatible across different versions of H3.
+ *
+ * @param origin An anchoring index for the ij coordinate system.
+ * @param index Index to find the coordinates of
+ * @param out ij coordinates of the index will be placed here on success
+ * @return 0 on success, or another value on failure.
+ */
+int H3_EXPORT(experimentalH3ToLocalIj)(H3Index origin, H3Index h3,
+                                       CoordIJ* out) {
+    // This function is currently experimental. Once ready to be part of the
+    // non-experimental API, this function (with the experimental prefix) will
+    // be marked as deprecated and to be removed in the next major version. It
+    // will be replaced with a non-prefixed function name.
+    CoordIJK ijk;
+    int failed = h3ToLocalIjk(origin, h3, &ijk);
+    if (failed) {
+        return failed;
+    }
+
+    ijkToIj(&ijk, out);
+
+    return 0;
+}
+
+/**
+ * Produces an index for ij coordinates anchored by an origin.
+ *
+ * The coordinate space used by this function may have deleted
+ * regions or warping due to pentagonal distortion.
+ *
+ * Failure may occur if the index is too far away from the origin
+ * or if the index is on the other side of a pentagon.
+ *
+ * This function is experimental, and its output is not guaranteed
+ * to be compatible across different versions of H3.
+ *
+ * @param origin An anchoring index for the ij coordinate system.
+ * @param out ij coordinates to index.
+ * @param index Index will be placed here on success.
+ * @return 0 on success, or another value on failure.
+ */
+int H3_EXPORT(experimentalLocalIjToH3)(H3Index origin, const CoordIJ* ij,
+                                       H3Index* out) {
+    // This function is currently experimental. Once ready to be part of the
+    // non-experimental API, this function (with the experimental prefix) will
+    // be marked as deprecated and to be removed in the next major version. It
+    // will be replaced with a non-prefixed function name.
+    CoordIJK ijk;
+    ijToIjk(ij, &ijk);
+
+    return localIjkToH3(origin, &ijk, out);
+}
+
+/**
+ * Produces the grid distance between the two indexes.
+ *
+ * This function may fail to find the distance between two indexes, for
+ * example if they are very far apart. It may also fail when finding
+ * distances for indexes on opposite sides of a pentagon.
+ *
+ * @param origin Index to find the distance from.
+ * @param index Index to find the distance to.
+ * @return The distance, or a negative number if the library could not
+ * compute the distance.
+ */
+int H3_EXPORT(h3Distance)(H3Index origin, H3Index h3) {
+    CoordIJK originIjk, h3Ijk;
+    if (h3ToLocalIjk(origin, origin, &originIjk)) {
+        // Currently there are no tests that would cause getting the coordinates
```

Comment:
What does this mean practically? Is there no way to create a new test case that would trigger this check to fail?

Reply:
As far as I can see there is no input to `h3ToLocalIjk` where the origin is the same as the index that would fail. The only cases that return failure involve either resolution mismatch, being on different base cells, or deleted subsequence on the pentagon (which should be unreachable when origin=index.)

Reply:
I see what you mean.